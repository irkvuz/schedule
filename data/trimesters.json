[
  {
    "IdTrimester": 0,
    "uYear": null,
    "dateStart": "0001-01-01T00:00:00",
    "dateFinish": "0001-01-01T00:00:00"
  },
  {
    "IdTrimester": 1069,
    "uYear": "2018-2019 уч.г.",
    "dateStart": "2018-09-01T00:00:00",
    "dateFinish": "2018-12-15T00:00:00"
  },
  {
    "IdTrimester": 1079,
    "uYear": "2018-2019 уч.г.",
    "dateStart": "2019-01-09T00:00:00",
    "dateFinish": "2019-05-18T00:00:00"
  },
  {
    "IdTrimester": 1108,
    "uYear": "2019-2020 уч.г.",
    "dateStart": "2019-09-01T00:00:00",
    "dateFinish": "2019-12-09T00:00:00"
  },
  {
    "IdTrimester": 1171,
    "uYear": "2020-2021 уч.г.",
    "dateStart": "2020-09-01T00:00:00",
    "dateFinish": "2020-12-08T00:00:00"
  },
  {
    "IdTrimester": 1219,
    "uYear": "2020-2021 уч.г.",
    "dateStart": "2021-01-09T00:00:00",
    "dateFinish": "2021-05-19T00:00:00"
  },
  {
    "IdTrimester": 1236,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2021-09-01T00:00:00",
    "dateFinish": "2021-12-08T00:00:00"
  },
  {
    "IdTrimester": 1282,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-10T00:00:00",
    "dateFinish": "2022-05-19T00:00:00"
  },
  {
    "IdTrimester": 1286,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-10T00:00:00",
    "dateFinish": "2022-02-28T00:00:00"
  },
  {
    "IdTrimester": 1287,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-06-21T00:00:00"
  },
  {
    "IdTrimester": 1290,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-03-29T00:00:00"
  },
  {
    "IdTrimester": 1291,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-05-17T00:00:00"
  },
  {
    "IdTrimester": 1293,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-10T00:00:00",
    "dateFinish": "2022-05-04T00:00:00"
  },
  {
    "IdTrimester": 1294,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-05-31T00:00:00"
  },
  {
    "IdTrimester": 1295,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-05-24T00:00:00"
  },
  {
    "IdTrimester": 1296,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-04-19T00:00:00"
  },
  {
    "IdTrimester": 1300,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-04-19T00:00:00"
  },
  {
    "IdTrimester": 1301,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-04-19T00:00:00"
  },
  {
    "IdTrimester": 1302,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-03-15T00:00:00"
  },
  {
    "IdTrimester": 1304,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-03-16T00:00:00",
    "dateFinish": "2022-06-21T00:00:00"
  },
  {
    "IdTrimester": 1305,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-04-12T00:00:00"
  },
  {
    "IdTrimester": 1306,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-05-03T00:00:00"
  },
  {
    "IdTrimester": 1307,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-05-10T00:00:00"
  }
]
