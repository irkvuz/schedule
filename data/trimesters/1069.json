{
  "IdTrimester": 1069,
  "uYear": "2018-2019 уч.г.",
  "dateStart": "2018-09-01T00:00:00",
  "dateFinish": "2018-12-15T00:00:00",
  "week": 1,
  "Error": null
}