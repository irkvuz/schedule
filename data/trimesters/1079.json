{
  "IdTrimester": 1079,
  "uYear": "2018-2019 уч.г.",
  "dateStart": "2019-01-09T00:00:00",
  "dateFinish": "2019-05-18T00:00:00",
  "week": 2,
  "Error": null
}