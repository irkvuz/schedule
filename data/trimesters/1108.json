{
  "IdTrimester": 1108,
  "uYear": "2019-2020 уч.г.",
  "dateStart": "2019-09-01T00:00:00",
  "dateFinish": "2019-12-09T00:00:00",
  "week": 1,
  "Error": null
}
