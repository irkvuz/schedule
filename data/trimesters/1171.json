{
  "IdTrimester": 1171,
  "uYear": "2020-2021 уч.г.",
  "dateStart": "2020-09-01T00:00:00",
  "dateFinish": "2020-12-08T00:00:00",
  "week": 1,
  "Error": null
}
