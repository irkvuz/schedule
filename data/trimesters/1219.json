{
  "IdTrimester": 1219,
  "uYear": "2020-2021 уч.г.",
  "dateStart": "2021-01-09T00:00:00",
  "dateFinish": "2021-05-19T00:00:00",
  "week": 1,
  "Error": null
}
