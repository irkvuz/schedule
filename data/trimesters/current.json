{
  "IdTrimester": 1236,
  "uYear": "2021-2022 уч.г.",
  "dateStart": "2021-09-01T00:00:00",
  "dateFinish": "2021-12-08T00:00:00",
  "week": 2,
  "Error": null
}
