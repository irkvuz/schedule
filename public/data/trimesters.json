[
  {
    "IdTrimester": 1287,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-06-21T00:00:00",
    "week": 2,
    "Error": null
  },
  {
    "IdTrimester": 1290,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-03-29T00:00:00",
    "week": 2,
    "Error": null
  },
  {
    "IdTrimester": 1291,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-05-17T00:00:00",
    "week": 2,
    "Error": null
  },
  {
    "IdTrimester": 1294,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-05-31T00:00:00",
    "week": 2,
    "Error": null
  },
  {
    "IdTrimester": 1295,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-05-24T00:00:00",
    "week": 2,
    "Error": null
  },
  {
    "IdTrimester": 1296,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-12T00:00:00",
    "dateFinish": "2022-04-19T00:00:00",
    "week": 2,
    "Error": null
  },
  {
    "IdTrimester": 1293,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-10T00:00:00",
    "dateFinish": "2022-05-04T00:00:00",
    "week": 2,
    "Error": null
  },
  {
    "IdTrimester": 1282,
    "uYear": "2021-2022 уч.г.",
    "dateStart": "2022-01-10T00:00:00",
    "dateFinish": "2022-05-19T00:00:00",
    "week": 2,
    "Error": null
  }
]
