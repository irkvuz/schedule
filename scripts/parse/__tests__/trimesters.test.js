import { getCurrentTrimesters } from "../trimesters.mjs";

describe('getCurrentTrimesters', () => {
  it('should parse and return ids', async ()=>{
    const trimesters = await getCurrentTrimesters();
    expect(trimesters).toMatchSnapshot();
  });
});
