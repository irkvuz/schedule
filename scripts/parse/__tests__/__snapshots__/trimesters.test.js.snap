// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCurrentTrimesters should parse and return ids 1`] = `
Array [
  1282,
  1286,
  1287,
  1290,
  1291,
  1294,
  1295,
]
`;
